File: bronze/1978/c++/main.cpp
#include <iostream>

int sqrt(int num)
{
    if (2 > num) { return 0; } 
    else if (num == 2) { return 1; }
    else if (num % 2 == 0) { return 0; }

    for (int i = 3; i * i <= num; i += 2) {
        if (num % i == 0) { return 0; }
    }

    return 1;
}

int main()
{
    int n, num, count = 0;
    std::cin >> n;
    for (int i = 0; i < n; i++) {
        std::cin >> num;
        count += sqrt(num);
    }
    std::cout << count;
    return 0;
}

File: bronze/1978/python/main.py

if (__name__ == '__main__'):
    n = int(input())
    num = list(map(int, input().split()))[:n]
    count = 0
    for data in num:
        if (data < 2): continue
        if (data == 2): count += 1
        if (data % 2 == 0): continue

        count += 1
        for i in range(3, int(data**0.5) + 1, 2):
            if (data % i == 0): count -= 1; break
    print(count)